barretenberg/cpp/src/barretenberg/vm/tests/AvmMini_execution.test.cpp

```diff
@@ -60,24 +61,28 @@ TEST_F(AvmMiniExecutionTests, basicAddReturn)
                                "00000000"; // ret size 0
 
     auto bytecode = hex_to_bytes(bytecode_hex);
-    auto instructions = Execution::parse(bytecode);
+    auto instructions = Deserialization::parse(bytecode);
 
     // 2 instructions
     EXPECT_EQ(instructions.size(), 2);
 
     // ADD
     EXPECT_EQ(instructions.at(0).op_code, OpCode::ADD);
-    EXPECT_EQ(instructions.at(0).operands.size(), 3);
-    EXPECT_EQ(instructions.at(0).operands.at(0), 7);
-    EXPECT_EQ(instructions.at(0).operands.at(1), 9);
-    EXPECT_EQ(instructions.at(0).operands.at(2), 1);
-    EXPECT_EQ(instructions.at(0).in_tag, AvmMemoryTag::U8);
+
+    auto operands = instructions.at(0).operands;
+    EXPECT_EQ(operands.size(), 4);
+    EXPECT_EQ(std::get<AvmMemoryTag>(operands.at(0)), AvmMemoryTag::U8);
+    EXPECT_EQ(std::get<uint32_t>(operands.at(1)), 7);
+    EXPECT_EQ(std::get<uint32_t>(operands.at(2)), 9);
+    EXPECT_EQ(std::get<uint32_t>(operands.at(3)), 1);
 
     // RETURN
     EXPECT_EQ(instructions.at(1).op_code, OpCode::RETURN);
-    EXPECT_EQ(instructions.at(1).operands.size(), 2);
-    EXPECT_EQ(instructions.at(1).operands.at(0), 0);
-    EXPECT_EQ(instructions.at(1).operands.at(0), 0);
+
+    operands = instructions.at(1).operands;
+    EXPECT_EQ(operands.size(), 2);
+    EXPECT_EQ(std::get<uint32_t>(operands.at(0)), 0);
+    EXPECT_EQ(std::get<uint32_t>(operands.at(1)), 0);
 
     auto trace = Execution::gen_trace(instructions, std::vector<FF>{});
```

Comment:
`std::vector<FF>` not needed. `{}` should be enough. (please check everywhere else in the modified files, there are other instances)

Also, BTW, I like this style which prefers to avoid braces initialization whenever possible: https://abseil.io/tips/88 . That's for an offline discussion though :)